xline/src/server/kv_server.rs

```diff
@@ -470,19 +467,34 @@ where
         let txn_req = request.get_ref();
         debug!("Receive grpc request: {:?}", txn_req);
         Self::check_txn_request(txn_req, self.kv_storage.revision())?;
-        let is_fast_path = false; // lock need revision of txn
-        let (cmd_res, sync_res) = self.propose(request, is_fast_path).await?;
 
-        let mut res = Self::parse_response_op(cmd_res.decode().into());
-        if let Some(sync_res) = sync_res {
-            let revision = sync_res.revision();
-            debug!("Get revision {:?} for TxnRequest", revision);
-            Self::update_header_revision(&mut res, revision);
-        }
+        let res = if txn_req.is_read_only() {
+            debug!("TxnRequest is read only");
+            let is_serializable = txn_req.is_serializable();
+            let token = get_token(request.metadata());
+            let wrapper = RequestWithToken::new_with_token(request.into_inner().into(), token);
+            let propose_id = self.generate_propose_id();
+            let cmd = Self::command_from_request_wrapper(propose_id, wrapper);
+            if is_serializable {
```

Comment:
should this be `if !is_serializable`?

Reply:
How come? We should do `wait_read_state` when a client sends us a serializable range request.

Reply:
Fixed， it should be `if !is_serializable`, we only need to do wait_read_state when the request is not a serializable request